curricula/self_paced/part_1/z230-1c.md

```diff
@@ -0,0 +1,245 @@
+
+6. [git] Clean up, stage, and commit (`git commit -m "Compare wind chill factors"`)
+
+7. [python] Now, format the output so that it's easy to 
+   understand and rename this script to something indicative 
+   of what it actually does.
+   
+   To the end of the file, add:
+   
+   ```python
+   # Output comparison of data
+   print('                ORIGINAL  COMPUTED')
+   print(' DATE    TIME  WINDCHILL WINDCHILL DIFFERENCE')
+   print('------- ------ --------- --------- ----------')
+   for date, time, wc_orig, wc_comp in zip(data['date'], data['time'], data['windchill'], windchill):
+       print(f'{date} {time:>6} {wc_orig:9.6f} {wc_comp:9.6f} {wc_orig-wc_comp:10.6f}')
+   ```
+   
+   Here you used f-string formatting with more `f-string` 
+   formatting options. `>6` indicates that you'd like the 
+   characters of the string to be right-justified and to take 
+   up 6 spaces. 
+   
+   `9f` specifies that you want the value to fill 9 spaces, 
+   so `9.6f` indicates you'd like the value to fill 
+   9 spaces with 6 of them being after the decimal point. 
+   Same concept for `10.6f`
+   
+   You now have your first complete Python script! 
+
+8. [git] DON'T CLEAN UP!  Just stage and commit
+   (`git commit -m "Output formatting comparison data"`)
+
+9. [git] Let's rename this script to something meaningful and 
+   indicative of the computation inside.
+
+   ```bash
+   $ git mv mysci.py windchillcomp.py
+   $ git commit -m "Renaming first script"
+   ```
+
+10. [git] Let's push to GitHub!
+   
+   First you have to create a remote repository.
+   
+   1. Go to https://github.com/ and create or login to your 
+   account.
+
+   2. At the top right of any Github page, there is a '+' icon. 
+   Click that, then select 'New Repository'.
+
+   3. Name your repository, "NCAR_python_tutorial_2020". 
+   It is best practice for your local project and GitHub 
+   repository to share a name.
+
+   4. And click "Create Repository" 
+
+   5. Copy the link to your GitHub repository. Typically this 
+   will have the form: Copy the link in the input right beneath 
+   the title, it should look something like this: 
+   "https://github.com/<user_name>/NCAR_python_tutorial_2020.git"
+
+   Then to set your remote repository, in your project terminal type:
+
+   ```bash
+   $ git remote add origin <remote repository URL>
+   ```
+
+   And verify your remote repository:
+
+   ```bash
+   $ git remote -v
+   ```
+   
+   And finally push your project to GitHub:
+
+   ```bash
+   $ git push origin master
+   ```
+
+   Think of GitHub as online storage for versions of your 
+   project, much like hosting your code in a Google Drive, 
+   but with better features specific to coding. A lot of 
+   GitHub's features show their usefulness when you are 
+   working collaboratively, sharing your code with other 
+   scientists, or if you wanted to display and easily 
+   visualize changes in your code between commits. 
+   
+==========================
```

Comment:
Again.  Not sure how best to render this, but maybe this won't be markdown in the end, anyway.